realm/realm-library/src/main/java/io/realm/RealmConfiguration.java

```diff
@@ -605,6 +620,9 @@ public Builder rxFactory(RxObservableFactory factory) {
          * @param transaction transaction to execute.
          */
         public Builder initialData(Realm.Transaction transaction) {
+            if (readOnly) {
```

Comment:
In e.g. `assetFile()` we do the check if `initialDataTransaction` is set and throw. I guess you can argue both ways. Throwing as soon as possible is nicer, but the Builder isn't technically "done" until you call `build()`.

For consistency, I would prefer to throw here for now as changing when `assetFile()` throws is technically a breaking change. I wouldn't be terribly opposed to changing it at the next major version bump though. Perhaps I should just create an issue for it?